test/studies/bale/indexgather/ig.chpl

```diff
@@ -0,0 +1,70 @@
+use CyclicDist;
+use BlockDist;
+use Random;
+use Time;
+
+config const printStats = true,
+             printArrays = false,
+             verify = true;
+
+config const useRandomSeed = true,
+             seed = if useRandomSeed then SeedGenerator.oddCurrentTime else 314159265;
+
+
+config const numTasksPerLocale = here.maxTaskPar;
+const numTasks = numLocales * numTasksPerLocale;
+config const N = 1000000; // number of updates per task
+config const M = 10000; // number of entries in the table perf task
+
+const numUpdates = N * numTasks;
+const tableSize = M * numTasks;
+
+// The intuitive implementation of indexgather that uses fine-grained GETs
+
+proc main() {
+  const Mspace = {0..tableSize-1};
+  const D = Mspace dmapped Cyclic(startIdx=Mspace.low);
+  var A: [D] int;
+
+  const Nspace = {0..numUpdates-1};
+  const D2 = Nspace dmapped Block(Nspace);
+  var rindex: [D2] int;
+
+  fillRandom(rindex, seed);
+  forall r in rindex {
+    r = mod(r, tableSize);
+  }
+
+  var tmp: [D2] int;
+  A = {0..tableSize-1};
```

Comment:
I think you can initialize ``A`` with this value, instead of assigning later. You could also probably just use a range.